src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -6704,6 +6720,705 @@ namespace HeatBalanceSurfaceManager {
+                WinHeatGain(surfNum) = 0.0;
+                WinHeatTransfer(surfNum) = 0.0;
+                WinHeatGainRep(surfNum) = 0.0;
+                WinHeatLossRep(surfNum) = 0.0;
+                WinGainConvGlazToZoneRep(surfNum) = 0.0;
+                WinGainIRGlazToZoneRep(surfNum) = 0.0;
+                WinLossSWZoneToOutWinRep(surfNum) = 0.0;
+                WinGainFrameDividerToZoneRep(surfNum) = 0.0;
+                WinGainConvGlazShadGapToZoneRep(surfNum) = 0.0;
+                WinGainConvShadeToZoneRep(surfNum) = 0.0;
+                OtherConvGainInsideFaceToZoneRep(surfNum) = 0.0;
+                WinGainIRShadeToZoneRep(surfNum) = 0.0;
+                SurfaceWindow(surfNum).FrameQRadOutAbs = 0.0;
+                SurfaceWindow(surfNum).FrameQRadInAbs = 0.0;
+                SurfaceWindow(surfNum).DividerQRadOutAbs = 0.0;
+                SurfaceWindow(surfNum).DividerQRadInAbs = 0.0;
+            }
+
+        // Calculate heat extract due to additional heat flux source term as the surface boundary condition - all HT surfaces
+            if (DataSurfaces::AnyHeatBalanceInsideSourceTerm) {
+                for (int surfNum = firstSurf; surfNum <= lastSurf; ++surfNum) {
+                    if (Surface(surfNum).InsideHeatSourceTermSchedule) {
+                        QAdditionalHeatSourceInside(surfNum) =
+                            EnergyPlus::ScheduleManager::GetCurrentScheduleValue(Surface(surfNum).InsideHeatSourceTermSchedule);
+                    }
+                }
+            }
+
+            // Set up coefficient arrays prior to calculations and precalc terms that do no change during iteration - non-window surfaces
+            int const firstNonWinSurf = Zone(zoneNum).NonWindowSurfaceFirst;
+            int const lastNonWinSurf = Zone(zoneNum).NonWindowSurfaceLast;
+            Real64 const timeStepZoneSeconds = TimeStepZoneSec; // local for vectorization
+            // this loop auto-vectorizes
+            for (int surfNum = firstNonWinSurf; surfNum <= lastNonWinSurf; ++surfNum) {
+
+                // Pre-calculate a few terms before the iteration loop
+                TempTermSurf(surfNum) = CTFConstInPart(surfNum) + QRadThermInAbs(surfNum) + QRadSWInAbs(surfNum) +
+                                        QAdditionalHeatSourceInside(surfNum) + HConvIn(surfNum) * RefAirTemp(surfNum) + QHTRadSysSurf(surfNum) +
+                                        QCoolingPanelSurf(surfNum) + QHWBaseboardSurf(surfNum) + QSteamBaseboardSurf(surfNum) +
+                                        QElecBaseboardSurf(surfNum) + (QRadSurfAFNDuct(surfNum) / timeStepZoneSeconds);
+                TempDivSurf(surfNum) =
+                    1.0 / (CTFInside0(surfNum) - IsAdiabatic(surfNum) * CTFCross0(surfNum) + IsPoolSurf(surfNum) * PoolHeatTransCoefs(surfNum) +
+                           IsNotPoolSurf(surfNum) * HConvIn(surfNum) + IterDampConst);
```

Comment:
Here's an example of using `IsPoolSurf` to include a term used for pools, and `IsNotPoolSurface` to include a term only used when it's not a pool. Looking at this now, perhaps this could simply be done by overriding `HConvIn` with `PoolHeatTransCoefs` elsewhere?  In any case, this is in a loop calculating `TempTermSurf(surfNum)` and `TempDivSurf(surfNum)` which don't change during the iteration loop below.